### DISCUSSION TITLE
What's the changes of creating Native Modules & Native UI Components?

### DISCUSSION THREAD

**Discussion by ezranbayantemur** [reactwg/react-native-new-architecture#18] [Q&A] [answered]
Currently official documentation (_next version_) doesn't say any change about creating native structures.

Will it be the same or will be any change?

And if it's gonna change, will it be minor things or major scale change to create native module & UI component?

**Comment by cortinico** [comment, marked as answer]
> Currently official documentation (next version) doesn't say any change about creating native structures.

If you're referring to:
https://reactnative.dev/docs/next/native-modules-intro
https://reactnative.dev/docs/next/native-components-android
https://reactnative.dev/docs/next/native-components-ios
and related pages, you're right.

Those pages are still referring to the old-architecture of React Native. We haven't updated them yet and we're going to update them in the near future.

Ideally we can start by adding a banner to the New Architecture section/migration guide.
In the long run, they will just be updated to use the New Architecture by default. Users can still access the older version of those pages if they wish.

> And if it's gonna change, will it be minor things or major scale change to create native module & UI component?

There will be some changes. Those pages explains how to adapt your old-arch module and components to the New Architecture:

https://reactnative.dev/docs/next/new-architecture-app-modules-android
https://reactnative.dev/docs/next/new-architecture-app-modules-ios
https://reactnative.dev/docs/next/new-architecture-app-renderer-android
https://reactnative.dev/docs/next/new-architecture-app-renderer-ios

Does this answer your question?

**Reply by ezranbayantemur** [reply]
Yes, currently these pages gives some information about new architecture implementation. I'll be check them out for uptade.

Thanks so much @cortinico!
